sql: fix telemetry panic for COPY

otan opened the pull request:
Resolves #88229

Release note (bug fix): Fixed a bug where if telemetry is enabled, COPY can sometimes cause the server to crash.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88322)
<!-- Reviewable:end -->

rafiss reviewed (approved):
lgtm, just left something to think about.

cucaroach reviewed (commented):
<img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88322#-:-NCVxL22E2rxpk6K8Xdb:bmxilld)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @cucaroach, @otan, and @rafiss)



<!-- Sent from Reviewable.io -->

otan commented on pkg/sql/exec_log.go:
```diff
@@ -402,6 +402,10 @@ func (p *planner) maybeLogStatementInternal(
 
 			stats = telemetryMetrics.getQueryLevelStats(stats)
+			var txnID string
+			if p.txn != nil {
```
i'll file a new PR for this and discuss with KV! going to unblock the release for it.

otan committed:
sql: fix telemetry panic for COPY

Copy could previously crash if telemetry is turned on and it has to be
emitted to the sampling log. This unfortunately was not caught in
earlier tests as it was not sampled - in which the tests have now been
updated to force sampling.

Release note (bug fix): Fixed a bug where if telemetry is enabled, COPY
can sometimes cause the server to crash.

otan commented:
bors r=rafiss,cucaroach
